specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-07-01/managedClusters.json

```diff
@@ -1226,6 +1226,60 @@
           }
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerService/managedClusters/{resourceName}/agentPools/{agentPoolName}/upgradeNodeImageVersion": {
```

Comment:
why cant this be enabled via a PATCH on agentPools/{name}. why do you need an explicit POST. I am assuming nodeImageVersion is a property on agentPools resource type

Reply:
or by doing adding a new property like updateImageVersion: "latest" or other such values.

Reply:
Because customer cannot specify a version in the request. We will always upgrade the version to the latest one.

Reply:
with input as "latest" and returned value as "AKSUbuntu-1804-2020.06.24", the API is not idempotent, which is one of the reason we changed to use POST to trigger the operation.

Reply:
But with POST, it wont be possible to invoke this or make the version change using a template. Which version will the API use to upgrade? Is it not going to be the latest? I dont see it taking a request body. 
Updating properties on a resource via a POST can break RBAC, so someone who does not have /write on this resource type can call this POST action to effect a change on the resource which could only have been possible via a PUT/PATCH.

Reply:
right, we do not support doing that through template after this because that design is not idempotent. Customer cannot specify a specific version, that means they do not know what value to PATCH and PUT. It makes this operation perfect for a POST opeartion.